src/Checkbox/CheckboxIcon.js

```diff
@@ -0,0 +1,42 @@
+// @flow
+
+import * as React from 'react';
+import { View } from 'react-native';
+import Icon from '../Icon';
+import StyleSheet from '../PlatformStyleSheet';
+import theme, { parsePxValue } from './styles';
+
+type Props = {|
+  +checked?: boolean,
+  +hasError?: boolean,
+  +disabled?: boolean,
+|};
+
+const styles = StyleSheet.create({
+  box: {
+    alignItems: 'center',
+    justifyContent: 'center',
+    borderWidth: 1,
+    borderStyle: 'solid',
+    borderColor: theme.orbit.borderColorCheckboxRadio,
+    backgroundColor: theme.orbit.backgroundInput,
+    borderRadius: parsePxValue(theme.orbit.borderRadiusNormal),
+    width: parsePxValue(theme.orbit.widthCheckbox),
+    height: parsePxValue(theme.orbit.heightCheckbox),
+  },
+  boxError: {
+    borderColor: theme.orbit.borderColorCheckboxRadioError,
+  },
+});
+
+export default function CheckboxIco({ checked, hasError, disabled }: Props) {
+  const errorState = hasError && !disabled && !checked;
+  const iconColor = disabled
+    ? theme.orbit.colorIconCheckboxRadioDisabled
+    : theme.orbit.colorIconCheckboxRadio;
+  return (
+    <View style={[styles.box, errorState && styles.boxError]}>
+      {checked && <Icon name="check" color={iconColor} size={14} />}
```

Comment:
I am not sure we should leave the size being given like this. Does 14 match the size of the font as well? If that's the case, we could take it from `defaultTokens`.

Reply:
Good point. Currently the size matches with font, but I don't think it will be rule all the time (because for example size of box is set by separate values).
In current `orbit-components` there is svg check icon with fixed 16px width and height. So it's currently also not really dynamic.
We can definitely consider to make it more responsive/dynamic, but it will probably require to extend `defaultTokens`.